pkg/controller/builder.go

```diff
@@ -185,3 +195,60 @@ func replaceImageHook(placeholder string, desiredImageEnvVar string) deploymentc
 		return []byte(newDeployment), nil
 	}
 }
+
+func updateEnv(con *corev1.Container, name, value string) ([]corev1.EnvVar, error) {
```

Comment:
It feels a bit confusing to me that we accept a pointer to a container we want to update (which means we can update it in place), but then we return a new slice where the caller is expected to update the container, but also `append` has interesting semantics around whether the backing array is updated in place (if it has capacity) vs if a new backing array is created (when it doesn't)

WDYT about changing the function to do things the following way?
```go
func updateEnv(con *corev1.Container, name, value string) error {
   // check presence, and return err
   con.Env = append(con.Env, corev1.EnvVar{name, value})
   return nil
}
```

Reply:
The original function was just taking in the env vars and modifying them and returning. I added the container to get the name, but didn't really consider the rest of the function...